backend/src/handlers_prelude/github_pull_request.rs

```diff
@@ -0,0 +1,54 @@
+//! Github Pull Request are sent here.
+use axum::{extract::State, http::StatusCode, Json, Router};
+use axum::routing::post;
+use tracing::{error, info};
+use serde_json::json;
+use crate::gh::create_pull_request;
+use crate::AppState;
+use color_eyre::Result;
+
+#[derive(serde::Deserialize, Debug)]
+pub struct CreatePRRequest {
+    pub head_branch: String,    // The branch containing changes
+    pub base_branch: String,     // The branch to merge into
+    pub title: String,
+    pub description: String,
+}
+
+pub async fn create_pull_request_handler(
+    State(state): State<AppState>,
+    Json(payload): Json<CreatePRRequest>,
+) -> Result<(StatusCode, Json<serde_json::Value>), (StatusCode, Json<serde_json::Value>)> {
+    info!("Received create pull request request: {:?}", payload);
+    let head_branch = payload.head_branch;
+    let base_branch = payload.base_branch;
+    let title = payload.title;
+    let description = payload.description;
+
+    // Get the access token
+    let token = state.gh_credentials.get(&state.reqwest_client).await.map_err(|err| {
+        error!("Failed to get GitHub token: {:?}", err);
+        (StatusCode::INTERNAL_SERVER_ERROR, Json(json!({"error": "Failed to get access token"})))
```

Comment:
The response json would ideally be more consistent, so the receiver doesn't need to have logic for which fields are/aren't present